pkg/didcomm/protocol/introduce/models.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package introduce
+
+import (
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/decorator"
```

Comment:
as an aside, is decorator a protocol or should it be under protocol/common?

@llorllale

Reply:
@troyronda  Do you mean to move `decorator` to pkg/didcomm/common/decorator ?

Reply:
that is the question i am posing.

Reply:
A decorator is not a protocol. From the [decorator RFC](https://github.com/hyperledger/aries-rfcs/blob/master/concepts/0011-decorators/README.md#tutorial) (emphasis mine):

> **A decorator is an optional chunk of JSON that conveys metadata.** Decorators are not declared in a core schema but rather supplementary to it. Decorators add semantic content broadly relevant to messaging in general, and not so much tied to the problem domain of a specific type of interaction.

Decorators don't have any of the key [ingredients](https://github.com/hyperledger/aries-rfcs/blob/master/concepts/0003-protocols/README.md#ingredients) of protocols.

However, since they are [adopted](https://github.com/hyperledger/aries-rfcs/blob/master/concepts/0003-protocols/template.md#adopted-messages) by other protocols (and some, like `~thread`, are integral to every protocol), I don't have a problem placing them inside the `protocol` directory.